lib/tasks/migrations/migrations.rake

```diff
@@ -121,6 +121,91 @@ namespace :migrations do
     exit has_encountred_issue
   end
 
+  task :room_settings, [:start, :stop] => :environment do |_task, args|
+    start, stop = range(args)
+    has_encountred_issue = 0
+
+    filtered_roles_ids = Role.unscoped
+                             .select(:id, :name)
+                             .where(name: COMMON[:filtered_user_roles])
+                             .pluck(:id)
+
+    Room.unscoped
+        .select(:uid, :room_settings)
+        .where.not(users: { role_id: filtered_roles_ids, deleted: true }, deleted: true)
+        .find_each(start: start, finish: stop, batch_size: COMMON[:batch_size]) do |r|
+
+      parsed_room_settings = JSON.parse(r.room_settings)
+      room_settings = {
+        record: parsed_room_settings["recording"],
+        muteOnStart: parsed_room_settings["muteOnStart"],
+        guestPolicy: parsed_room_settings["requireModeratorApproval"],
+        glAnyoneCanStart: parsed_room_settings["anyoneCanStart"],
+        glAnyoneJoinAsModerator: parsed_room_settings["joinModerator"],
+        glViewerAccessCode: r.access_code,
+        glModeratorAccessCode: r.moderator_access_code
+      }
```

Comment:
I believe the logic here is solid and will answer the feature requests,
And the keys mapping is correct, well done!
However, there's few things to note, so that not all the values in v2 room settings are directly migratable to v3, :
1. Some values needs to be mapped for compatibility:
    - All values on v2 room settings are Booleans verses being strings on v3 (rails should be able to cast that automatically on v3 but we may need more control over this to reduce error surface.).
  2. Room meetings options values on v3 aren't binary meaning there's more then 2 possible values unlike on v2, e.g. `"requireModeratorApproval" => true` on v2 maps to `"guestPolicy" => "ALWAYS_ACCEPT"` and not "guestPolicy" => "true"`.
 3. IMO Access codes may be better to be left to be generated by v3, on v2 they were generated insecurely which was patched on v3.
 Migrating insecure codes to v3 will break the security mechanism it introduced for generating access codes until someone decides to regenerate codes (which since we allow them to migrate old used codes they won't bather changing it).

Reply:
Here's an example of a room settings Json object on v2:
```json
{"muteOnStart":true,"requireModeratorApproval":true,"anyoneCanStart":true,"joinModerator":true,"recording":false}
```

Reply:
1 is in progress
2 Great call. will purge Access Codes. Users will have to re-generate an Access Code if needed.